### PR TITLE
feature: add reusable QA component

### LINKED ISSUE #61: Q/A Component
Select only 1 or multiple select

### PR DESCRIPTION
**Pull request by Fayzullo05**
1. **Summary:** _What changed, and why?_
  - Implemented reusable QA component for rendering questions with answer options which supports both single-choice and multiple-choice questions.
----
 - Add requested changes
 - Add tests

3. **Linked Issue:** Closes #61 

4. **UI Screenshots:** image

---

## Definition of Done (DoD)

Before requesting review, confirm that:

- [x] Acceptance criteria are satisfied
- [x] Code follows the team's style guide and conventions
- [x] No type assertions are used
- [x] No explicit `any` types are used
- [x] No debug logs or commented-out code remain
- [x] ESLint and formatting checks pass
- [x] No console errors or warnings
- [x] Tests were added or updated if necessary
- [x] All existing tests pass locally
- [x] Edge cases and error states are handled
- [x] Pull request is linked to an issue

### CODE REVIEW

File: frontend/src/components/shared/QA/types.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  question: string;
+  options: QAOption[];
+  multiple?: boolean;
+};
+
+export type QAProps = {
+  id: string;
+  question: string;
+  options: QAOption[];
+  multiple?: boolean;
```
[the remark this answers is not in the document]
Comment (Fayzullo05) [reply frontend/src/components/shared/QA/types.ts]: Сделаю изменения и добавлю в PR
